knitr/convert-odes/convert_odes.Rmd

```diff
@@ -0,0 +1,398 @@
+
+4. ```times``` - Sorted array of times to which the ode will be solved (each
+  element must be greater than t0)
+
+5. ```arg1, arg2, ...``` - Arguments passed unmodified to the ODE system
+function. There can be any number of ```...``` arguments, and the ```...```
+arguments can be any type, but they must match the types of the corresponding
+```...``` arguments of ```f```.
+
+The ODE system function should take the form:
+
+```{stan, output.var = "", eval = FALSE}
+vector f(real t, vector y, T1 arg1, T2 arg2, ...)
+```
+
+The arguments are:
+
+1. ```t``` - Time at which to evaluate the ODE system function
+
+2. ```y``` - State at which to evaluate the ODE system function
+
+3. ```arg1, arg2, ...``` - Arguments passed unmodified from the ODE solver
+function call. The ```...``` must match the types of the corresponding
+```...``` arguments of the ODE solver function call.
+
+A call to `ode_bdf` returns the solution of the ODE specified by the right hand
+side (`f`) and the initial conditions (`y0` and `t0`) at the time points given
+by the `times` argument. The solution is given by an array of vectors.
+
+The proposed `ode_bdf_tol` interface is (the interfaces for `ode_rk45_tol`
+and `ode_adams_tol` are the same):
+
+```{stan, output.var = "", eval = FALSE}
+vector[] ode_bdf_tol(F f,
+                     vector y0,
+                     real t0, real[] times,
+                     real rel_tol, real abs_tol,
+                     int max_num_steps,
+                     ...)
+```
+
+The arguments are:
+1. ```f``` - ODE system function
+
+2. ```y0``` - Initial state of the ODE
+
+3. ```t0``` - Initial time of the ODE
+
+4. ```times``` - Sorted array of times to which the ode will be solved (each
+  element must be greater than t0)
+
+5. ```rel_tol``` - Relative tolerance for solve (data)
+
+6. ```abs_tol``` - Absolute tolerance for solve (data)
+
+7. ```max_num_steps``` - Maximum number of timesteps to take in integrating
+  the ODE solution between output time points (data)
+
+5. ```...``` - Arguments passed unmodified to the ODE system
+function. There can be any number of ```...``` arguments, and the ```...```
+arguments can be any type, but they must match the types of the corresponding
+```...``` arguments of ```f```.
+
+The `ode_rk45`/`ode_bdf`/`ode_adams` interfaces are just wrappers around the
+`ode_rk45_tol`/`ode_bdf_tol`/`ode_adams_tol` interfaces with defaults for
+`rel_tol`, `abs_tol`, and `max_num_steps`. For the RK45 solver the defaults
+are $10^{-6}$ for `rel_tol` and `abs_tol` and $10^6$ for `max_num_steps`.
+For the BDF/Adams solvers the defaults are $10^{-10}$ for `rel_tol` and
+`abs_tol` and $10^8$ for `max_num_steps`.
+
+For more detailed information about either interface, look at the function
+reference guide:
+[Old interface](https://mc-stan.org/docs/2_23/functions-reference/functions-ode-solver.html),
+[New interface](https://mc-stan.org/docs/2_23/functions-reference/functions-ode-solver.html)
```

Comment:
I think the links need an update (I don't know why this comment refers to many lines, I mean 177).

Reply:
These links will be off until the 2.24 docs get built.